airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java

```diff
@@ -541,13 +559,77 @@ public interface Configs {
    */
   boolean shouldRunConnectionManagerWorkflows();
 
+  /**
+   * Define if the worker is operating as a Control Plane worker, a Data Plane worker, or as a
+   * combined worker. - Control plane workers process tasks related to control-flow, like scheduling
+   * and routing. - Data plane workers process tasks related to data syncing, and are typically
+   * deployed outside of Airbyte's internal network. - Combined workers process all tasks, and are the
+   * default for OSS deployments. - Internal-use only.
+   */
+  WorkerPlane getWorkerPlane();
+
+  // Worker - Control Plane configs
+
+  /**
+   * Return true if the WorkerPlane is CONTROL_PLANE or COMBINED. Internal-use only.
+   */
+  boolean isControlPlaneWorker();
+
+  /**
+   * TEMPORARY: Define a set of connection IDs that should run in Airbyte's AWS Data Plane. - This
+   * should only be set on Control-plane workers, since those workers decide which Data Plane task
+   * queue to use based on connectionId. - Will be removed in favor of the Routing Service in the
+   * future. Internal-use only.
+   */
+  Set<String> connectionIdsForAwsDataPlane();
+
+  /**
+   * Define the task queue where the Control Plane worker will route Data Plane tasks to by default.
+   * Internal-use only.
+   */
+  String getDefaultDataPlaneTaskQueue();
+
+  // Worker - Data Plane configs
+
+  /**
+   * Return true if the WorkerPlane is DATA_PLANE or COMBINED. Internal-use only.
+   */
+  boolean isDataPlaneWorker();
+
+  /**
+   * Define a set of Temporal Task Queue names for which the Data Plane worker should register
+   * handlers for. Entries should correspond with the Data Plane where the worker is deployed.
+   * Internal-use only.
+   */
+  Set<String> getDataPlaneTaskQueues();
+
+  /**
+   * Return the control plane endpoint a data plane will hit. This is used for authentication and
+   * separate from the actual endpoint being hit. Internal-use only.
+   */
+  String getControlPlaneGoogleEndpoint();
```

Comment:
nit: can we call this `getControlPlaneAuthEndpoint` to minimise leaking Cloud bits?